apps/studio/prisma/scripts/addUsersToSite.ts

```diff
@@ -0,0 +1,45 @@
+import cuid2 from "@paralleldrive/cuid2"
+
+import { db, RoleType } from "~/server/modules/database"
+
+// NOTE: add them as editor first as that is the one with the least permissions
+export const addUsersToSite = async (siteId: number, emails: string[]) => {
+  const lowercasedEmails = emails.map((email) => email.toLowerCase())
+
+  await Promise.all(
+    lowercasedEmails.map(async (email) => {
+      await db.transaction().execute(async (tx) => {
+        const user = await tx
+          .insertInto("User")
+          .values({
+            id: cuid2.createId(),
+            name: email.split("@")[0] || "",
+            email,
+            phone: "",
+          })
+          .onConflict((oc) =>
+            oc
+              .column("email")
+              .doUpdateSet((eb) => ({ email: eb.ref("excluded.email") })),
+          )
+          .returning(["id", "name"])
+          .executeTakeFirstOrThrow()
+
+        await tx
+          .insertInto("ResourcePermission")
+          .values({
+            userId: user.id,
+            siteId,
+            role: RoleType.Editor,
```

Comment:
nit: might want to customise as part of the `emails` variable too?

Reply:
honestly, can - i just did this first because the requests we were getting were all to add users as `Admin`, which would be easy to do with the followup script. we didn't really see instances w mixed perms but better to have and set as default arg i g